pkg/client/kubeclient.go

```diff
@@ -73,6 +68,29 @@ func newSchedulerKubeClient(kc string) SchedulerKubeClient {
 	}
 }
 
+func createRestConfig(kc string) *rest.Config {
+	// attempt to use in-cluster config
+	config, err := rest.InClusterConfig()
+	if err != nil && err != rest.ErrNotInCluster {
+		// this is called during early initialization, so abort
+		panic(err)
```

Comment:
Nit: how about `log.Logger.Panic()` or `log.Logger.Fatal()`?

Reply:
Circular dependencies. This is called before logging is safe to call. Panic will log the error and abort, so we still get it. It's also pretty unlikely to fail in real life.

Reply:
Hmm, noticed after commit that we can in fact do this. Will update in the next PR in this series.